duckduckgo/iOS
# Autofill change to sorting algorithm for displaying logins

Task/Issue URL: https://app.asana.com/0/0/1203927830097167/f
Tech Design URL:
CC: @THISISDINOSAUR 

**Description**:
Changes to Autofill logins management screen taking subdomains and ports into account



**Steps to test this PR**:

**URL store changes** https://app.asana.com/0/0/1203927830097159/f, https://app.asana.com/0/1201645688934642/1203927830097162/f
1.  Go to https://fill.dev/form/login-simple
2. Login and store the credentials
3. Check the website URL saved is `fill.dev`
4. Add a new login manually and set the Website URL to something like `https://www.example.com?params=123`
5. Save and check the website URL saved is `www.example.com`
6. Add a new login manually with a port, setting the Website URL to something like `https://www.example.com:123`
7. Save and check the website URL saved is `www.example.com:123`

**URL matching changes** 
1. Add a login for `fill.dev`
2. Visit `fill.dev/form/login-simple` and confirms it prompts to autofill as normal
3. Open Autofill login list screen; verify it is listed as a suggested site
4. Edit the login and append a port to the URL (e.g., `fill.dev:9000`)
6. Visit the login page again (refreshing if required); verify you are not prompted to autofill (as ports don't match)
7. Open Autofill login list screen; confrim it is not listed as a suggested site

**URL sorting changes** https://app.asana.com/0/0/1203927830097167/f
1. Manually add the login entries detailed in https://app.asana.com/0/0/1203927830097167/f
2. In the Autofill login screen, confirm the logins are sorted in the same order as in the task
3. Add any other combinations of titles, domains and subdomains to test the sort order matches expectations 

**Leave title blank by default changes** https://app.asana.com/0/0/1203927830097163/f
1. Create a new login, setting only the Website URL, leaving the title blank. 
2. Once saved the title should be the Website URL
3. Tap to Edit the saved login and the title should display placeholder text 'Title'
4. Leave the title blank and save. Confirm the title still displays the Website URL and in edit mode still displays placeholder text 'Title'
5. Edit the login and set a title. Save and confirm the login is displaying the newly set title.
6. Save a new login via a website login save prompt. Tap to view the saved login
7. Confirm the title is displaying placeholder text 'Title' by default when in edit mode8. 

**Change to Favicon behavior** https://app.asana.com/0/0/1204031392481537/f
1. Create a login with a title that is different to the domain of the website
2. Confirm the placeholder letter is the first letter of the title, not of the domain
3. Create a login with a subdomain, but no title
4. Confirm the placeholder letter is the first letter of the domain, not of the subdomain
5. Confirm that otherwise favicons behave as before

**Changes to login prompt screen to illustrate partial subdomain matches** https://app.asana.com/0/0/1203927830097168/f
1. Create logins following the examples in the [figma design](https://www.figma.com/file/aM8i2nPES4Edf0EGXKhnwI/Autofill-(iOS)?node-id=2257%3A26485&t=7On7AyV4C09UdGwz-1)
2. Confirm the logins display and group correctly, including the More Options button 

**Copy updates (Translations TBD)**
1. Confirm copy update as per https://app.asana.com/0/0/1204129221412822/f
2. Confirm copy update as per https://app.asana.com/0/0/1203965896800930/f



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
